api/src/login/login.service.ts

```diff
@@ -0,0 +1,24 @@
+import bcrypt from "bcrypt";
+import { IUser, UserModel } from '../user/user.model';
+import jwt from 'jsonwebtoken';
+import config from '../config';
+
+let err: any;
+
+export const loginService = {
+    async login(user: IUser) {
```

Comment:
A type-okat sokkal specifikusabban kell használnod: itt nem egy teljes `IUser`-t vársz paraméterként, hanem csak egy email-t és password-öt. Erre nyugodtan hozz létre egy új típust pl `LoginInputParams` vagy valami ilyesmi néven.

A másik lehetőség, hogy in-line definiálod a típust:
```
async login (user: { email: string, password: string })
```